script/munin-run

```diff
@@ -77,78 +74,81 @@ sub main
-            "servicedir=s" => \$servicedir,
-            "sconfdir=s"   => \$sconfdir,
-            "sconffile=s"  => \$sconffile,
-            "paranoia!"    => \$paranoia,
-            "version"      => \&print_version_and_exit,
-            "help"         => \&print_usage_and_exit,
-    );
+    my ( $plugin, $arg );
+
+    print_usage_and_exit()
+      unless GetOptions(
+        "config=s"     => \$conffile,
+        "debug!"       => \$DEBUG,
+        "verbose!"     => \$VERBOSE,
+        "pidebug!"     => \$PIDEBUG,
+        "servicedir=s" => \$servicedir,
+        "sconfdir=s"   => \$sconfdir,
+        "sconffile=s"  => \$sconffile,
+        "paranoia!"    => \$paranoia,
+        "version"      => \&print_version_and_exit,
+        "help"         => \&print_usage_and_exit,
+      );
 
     print_usage_and_exit() unless $ARGV[0];
 
     # Detaint the plugin name
-    ($plugin) = ($ARGV[0] =~ m/^([-\w.:]+)$/) or die "# ERROR: Invalid plugin name '$ARGV[0].\n";
-    if ($ARGV[1]) {
-        ($arg) = ($ARGV[1] =~ m/^(\w+)$/)
-            or die "# ERROR: Invalid characters in argument '$ARGV[1]'.\n";
+    ($plugin) = ( $ARGV[0] =~ m/^([-\w.:]+)$/x )
+      or die "# ERROR: Invalid plugin name '$ARGV[0]'.\n";
+    if ( $ARGV[1] ) {
+        ($arg) = ( $ARGV[1] =~ m/^(\w+)$/x )
+          or die "# ERROR: Invalid characters in argument '$ARGV[1]'.\n";
+        die "# ERROR: Invalid plugin argument '$ARGV[1]'.\n"
+          unless grep( /^$ARGV[1]$/x, qw(autoconf config suggest) );
     }
 
-    # Detaint service directory.  FIXME: do more strict detainting?
+    # Detaint service directory.
     if ($servicedir) {
-        $servicedir =~ /(.*)/;
-        $servicedir = $1;
+        die "# ERROR: Invalid servicedir supplied.\n" unless -d $servicedir;
+        $servicedir =~ /(.*)/x;
+        $servicedir = $1;    ## no critic qw(RegularExpressions::ProhibitCaptureWithoutTest)
     }
 
     # Update the config
-    $config->reinitialize({
-        %$config,
-
-        sconfdir   => $sconfdir,
-        conffile   => $conffile,
-        sconffile  => $sconffile,
-        VERBOSE    => $VERBOSE,
-        DEBUG      => $DEBUG,
-        paranoia   => $paranoia,
-    });
-
-    return ($plugin, $arg);
+    $config->reinitialize( {
+            %{$config},      ## no critic qw(ValuesAndExpressions::ProhibitCommaSeparatedStatements)
+
+            sconfdir  => $sconfdir,
+            conffile  => $conffile,
+            sconffile => $sconffile,
+            VERBOSE   => $VERBOSE,
+            DEBUG     => $DEBUG,
+            paranoia  => $paranoia,
+    } );
+
+    return ( $plugin, $arg );
 }
 
 
-sub print_usage_and_exit
-{
+sub print_usage_and_exit {
     require Pod::Usage;
-    Pod::Usage::pod2usage(-verbose => 1);
+    Pod::Usage::pod2usage( -verbose => 1 );
 }
 
 
-sub print_version_and_exit
-{
+sub print_version_and_exit {
     require Pod::Usage;
     Pod::Usage::pod2usage(
-        -verbose => 99,
+        -verbose  => 99,
```

Comment:
Is this alignment of operators really a common style recommendation?
(Python's [PEP8](https://www.python.org/dev/peps/pep-0008/) defines exactly the opposite)

Reply:
It's the default perltidy setting

Reply:
I used to be very fond of it, as it makes the code a little more readable. But it has to be automated, which is okay with `perltidy`. 

But what I prefer nowadays is the cleanliness of diffs, and usually this adds a lot of whitespace noise. It also makes `git blame` a headache to work with. Which is a tool that I use a lot ot understand the context of when the current code was written. 

This is invaluable to be able to patch things efficiently according to the original intend. Moreover, code that I wrote 6 months ago feels just like code someone else wrote 😉 so...

Reply:
Instead of `git blame` you could use [`git log -p -M --follow --stat -- path/to/your/file`](https://blog.andrewray.me/a-better-git-blame/)

Reply:
Err... Not really.

I don't really want to spend brain cycles on something that could be easily automated for me ;)

That said, a [git blame can be enhanced](https://jfire.io/blog/2012/03/07/code-archaeology-with-git/) to cope with it. But let's do it once for all just prior to releasing.

Reply:
Anyway: thank you both for sharing your hints :)

What shall we do about the alignment of operators?

Reply:
> What shall we do about the alignment of operators?
Let's defer all tidy changes to when we'll enforce the usage of perltidy.